proposals/0400-init-accessors.md

```diff
@@ -491,13 +491,22 @@ init() {
 }
 ```
 
+### Restrictions
+
+* A property with an `init` accessor must be declared in the same source file 
+  as the respective type declaration.
+* An `init` accessor is always `fileprivate` and thus cannot be called from
+  within a different source file.
+
 ## Source compatibility
 
 `init` accessors are an additive capability with new syntax; there is no impact on existing source code.
 
 ## ABI compatibility
 
-`init` accessors are only called from within a module, so they are not part of the module's ABI. In cases where a type's initializer is `@inlinable`, the body of an `init` accessor must also be inlinable.
+`init` accessors are only called from within a source file, so they are not part of the module's ABI.
```

Comment:
I think the original text is correct. Even if you can only define init accessors in the primary declaration of a type, you can add an initializer in another source file that will call the init accessor:

```swift
// in FirstFile.swift

struct S {
  var _x: Int

  var x: Int {
    @storageRestrictions(initializes: _x)
    init {
      self._x = newValue
    }
    get { _x }
  }
}

// in SecondFile.swift

extension S {
  init() {
    self.x = 10
  }
}
```

Reply:
Thanks. We had a crash report with like the simplest example of an init accessor use across files, and I was not sure if this sentence really meant they could be `internal` because the forum topics appeared to settle on `fileprivate`. I could have just inferred it from the `inlinable` business that follows, really. But I still think this paragraph would benefit from a little more straightforwardness and accuracy (see the latest revision).